src/Tools/PrepareTests/MinimizeUtil.cs

```diff
@@ -40,59 +60,45 @@ void initialWalk()
             directories = directories.Concat(Directory.EnumerateDirectories(artifactsDir, "*.UnitTests"));
             directories = directories.Concat(Directory.EnumerateDirectories(artifactsDir, "RunTests"));
 
-            foreach (var unitDirPath in directories)
+            var idToFilePathMap = directories.AsParallel()
+                .SelectMany(walkDirectory)
+                .GroupBy(pair => pair.mvid)
+                .ToDictionary(
+                    group => group.Key,
+                    group => group.Select(pair => pair.pathInfo).ToList());
+
+            return idToFilePathMap;
+        }
+
+        IEnumerable<(Guid mvid, FilePathInfo pathInfo)> walkDirectory(string unitDirPath)
```

Comment:
This method is called in parallel yet it's not `static` and didn't add any data synchronization primitives. That makes me suspicious.

Reply:
Have made the captures explicit to make it a little easier to have confidence in this 😉